personalized_active_learning/datasets/base.py

```diff
@@ -0,0 +1,633 @@
+
+        """
+
+    # TODO: Remove, will be used in load_data_and_annotations
+    @property
+    @abc.abstractmethod
+    def data_file_relative_path(self) -> str:
+        """Get the relative path to csv file containing data.
+
+        `self.data_dir / self.data_file_relative_path` should point
+        to the csv file containing data.
+
+        Returns:
+            the relative path to csv file containing data.
+
+        """
+
+    # TODO: Remove, will be used in load_data_and_annotations
+    @property
+    @abc.abstractmethod
+    def data_dir(self) -> Path:
+        """Get the path to directory containing data.
+
+        Returns:
+            The path to directory containing data.
+
+        """
+
+    @abc.abstractmethod
+    def load_data_and_annotations(self) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """Load data (texts) and annotations.
+
+        Additional preprocessing & filtering should go here.
+
+        Returns:
+            Tuple of texts and annotations.
+
+        """
+
+    @property
+    def annotators_number(self) -> int:
+        """Get the number of annotators.
+
+        Returns:
+            The number of annotators.
+
+        """
+        return max(self.annotations["annotator_id"]) + 1
+
+    @property
+    def num_annotations(self) -> int:
+        """Get the number of annotations.
+
+        Returns:
+            The number of annotations.
+
+        """
+        return len(self.annotations)
+
+    @property
+    def train_folds_indices(self) -> List[int]:
+        """Get the list of training folds' indices.
+
+        Returns:
+            The list of training folds' indices.
+
+        """
+        all_folds = range(self.folds_num)
+        exclude_folds = [self.val_fold_index, self.test_fold_index]
+
+        return [fold for fold in all_folds if fold not in exclude_folds]
+
+    @property
+    def val_fold_index(self) -> int:
+        """Get the index of validation fold.
+
+        Returns:
+            The index of validation fold.
+
+        """
+        return (self._test_fold_index + 1) % self.folds_num
+
+    @property
+    def test_fold_index(self) -> int:
+        return self._test_fold_index
+
+    # TODO: Rewrite embeddings
+    @property
+    def text_embedding_dim(self) -> int:
+        if not self.embeddings_type in EMBEDDINGS_SIZES:
+            raise NotImplementedError()
+
+        return EMBEDDINGS_SIZES[self.embeddings_type]
+
+    @property
+    def annotations_with_data(self) -> pd.DataFrame:
+        return self.annotations.merge(self.data)
+
+    def __init__(
+        self,
+        batch_size: int = 3000,
+        split_mode: SplitMode = SplitMode.TEXTS,
+        embeddings_type: str = "labse",  # TODO: Change to class
+        major_voting: bool = False,  # TODO: Not sure if we need that
+        test_major_voting: bool = False,  # TODO: Not sure if we need that
+        folds_num: int = 10,  # TODO: Can be obtained from data?
```

Comment:
**[Comment]** I think that `folds_num` should be a experiment's parameter - like it is now.

Reply:
Right. Removed comment!